app/src/main/java/ch/epfl/sweng/studyup/teacher/ManageCourseActivity.java

```diff
@@ -0,0 +1,289 @@
+import static ch.epfl.sweng.studyup.utils.Constants.FB_COURSE_REQUESTS;
+import static ch.epfl.sweng.studyup.utils.Constants.FB_FIRSTNAME;
+import static ch.epfl.sweng.studyup.utils.Constants.FB_LASTNAME;
+import static ch.epfl.sweng.studyup.utils.Constants.FB_REQUESTED_COURSES;
+import static ch.epfl.sweng.studyup.utils.Constants.Course.FakeCourse;
+import static ch.epfl.sweng.studyup.utils.Constants.FB_USERS;
+import static ch.epfl.sweng.studyup.utils.Constants.INITIAL_FIRSTNAME;
+import static ch.epfl.sweng.studyup.utils.Constants.INITIAL_LASTNAME;
+import static ch.epfl.sweng.studyup.utils.Constants.INITIAL_SCIPER;
+import static ch.epfl.sweng.studyup.utils.GlobalAccessVariables.MOCK_ENABLED;
+
+public class ManageCourseActivity extends NavigationTeacher{
+    private List<CourseRequest> requests;
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_manage_courses);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        getAllRequests();
+    }
+
+    public void setupListViews(){
+        List<Course> otherCourses = new ArrayList<>(Arrays.asList(Course.values()));
+        otherCourses.remove(FakeCourse);
+        if(MOCK_ENABLED) otherCourses = new ArrayList<>(Arrays.asList(FakeCourse));
+
+        CourseRequest testRequest = new CourseRequest(FakeCourse, INITIAL_SCIPER, INITIAL_FIRSTNAME, INITIAL_LASTNAME);
+        if(MOCK_ENABLED) {
+            if (requests.contains(testRequest))
+                requests = new ArrayList<>(Arrays.asList(testRequest));
+            else
+                requests = new ArrayList<>();
+        }
+
+        List<Course> pendingCourses = new ArrayList<>();
+        for(CourseRequest request : requests){
+            if(Player.get().getSciperNum().equals(request.sciper))
+                pendingCourses.add(request.course);
+        }
+
+        List<Course> teachingCourses = Player.get().getCoursesTeached();
+
+        otherCourses.removeAll(pendingCourses);
+        otherCourses.removeAll(teachingCourses);
+
+        //Other courses
+        ((NonScrollableListView) findViewById(R.id.listViewOtherCourses)).setAdapter(new ListCourseAdapter(this, otherCourses, R.layout.model_course_send_request, false));
+
+        //Pending courses
+        if(Player.get().isSuperUser()){
+            ((NonScrollableListView) findViewById(R.id.listViewPendingCourses)).setAdapter(new requestListViewAdapter(this, requests));
+        }else{
+            ((NonScrollableListView) findViewById(R.id.listViewPendingCourses)).setAdapter(new ListCourseAdapter(this, pendingCourses, R.layout.model_course_pending, false));
+        }
+
+        //Accepted courses
+        ((NonScrollableListView) findViewById(R.id.listViewAcceptedCourses)).setAdapter(new ListCourseAdapter(this, teachingCourses, R.layout.model_course_accepted, false));
+    }
+
+    public void getAllRequests() {
```

Comment:
Why not in the Firestore class ?

Reply:
The problem with having Firestore class that have those method is that code climate is unhappy with this (that's mostly why we have a not-so-good complexity). So I decided to stop putting all functions regarding Firestore in the same class, as it is already >150 lines too much for code climate. I think it makes even more sens like this, since Firestore's instance is a singleton and that we use those two functions only here.

I pushed this approach quite further with the second PR that will come tomorrow, as I moved all functions like this (that is if they are used just (or almost) in one class) that are in Firestore.java in the classes where they are used.